x-pack/platform/plugins/shared/cases/server/client/cases/utils.test.ts

```diff
@@ -1073,6 +1075,183 @@ describe('utils', () => {
     });
   });
 
+  describe('getInProgressInfoForUpdate', () => {
+    const date = '2021-02-03T17:41:26.108Z';
+
+    it('returns the correct in_progress_at info when the case is marked as in-progress', async () => {
+      expect(
+        getInProgressInfoForUpdate({ status: CaseStatuses['in-progress'], updatedAt: date })
+      ).toEqual({
+        in_progress_at: date,
+      });
+    });
+
+    it('should not reset in_progress_at when the case is reopened', async () => {
+      expect(
+        getInProgressInfoForUpdate({ status: CaseStatuses.open, updatedAt: date })
+      ).toBeUndefined();
+    });
+
+    it('returns undefined if the status is not provided', async () => {
+      expect(getInProgressInfoForUpdate({ updatedAt: date })).toBeUndefined();
+    });
+  });
+
+  describe('getTimingMetricsForUpdate', () => {
+    const createdAt = '2021-11-23T19:00:00Z';
+    const inProgressAt = '2021-11-23T19:00:10Z';
+    const updatedAt = '2021-11-23T19:00:20Z';
+
+    it('should return the correct time_to_acknowledge when the case is marked as in-progress', () => {
+      expect(
+        getTimingMetricsForUpdate({
+          status: CaseStatuses['in-progress'],
+          createdAt,
+          updatedAt,
+        })
+      ).toEqual({
+        time_to_acknowledge: 20,
+        time_to_investigate: null,
+        time_to_resolve: null,
+      });
+    });
+
+    it('should not overwrite time_to_acknowledge', () => {
```

Comment:
```suggestion
    it('should not overwrite time_to_acknowledge when <some reason here>', () => {
```

Same for the other tests.

Reply:
Done ✅